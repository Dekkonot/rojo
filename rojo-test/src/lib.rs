use std::{
    fs,
    path::Path,
    process::Command,
};

use insta::assert_snapshot_matches;
use tempfile::tempdir;

static BUILD_TESTS: &[&str] = &[
    "gitkeep",
    "txt_in_folder",
];

#[test]
fn build_tests() {
    let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
    let build_test_path = manifest_dir.join("build-tests");
    let working_dir = manifest_dir.parent().unwrap();

    let output_dir = tempdir().expect("couldn't create temporary directory");

    for &test_name in BUILD_TESTS {
        let input_path = build_test_path.join(test_name);
        let output_path = output_dir.path().join(format!("{}.rbxmx", test_name));

        let status = Command::new("cargo")
            .args(&[
                "run", "--",
                "build", input_path.to_str().unwrap(), "-o", output_path.to_str().unwrap(),
            ])
            .current_dir(working_dir)
            .status()
            .expect("Couldn't start Rojo");

        assert!(status.success(), "Rojo did not exit successfully");

        let contents = fs::read_to_string(&output_path)
            .expect("Couldn't read output file");

        assert_snapshot_matches!(format!("build_{}", test_name), contents);
    }
}