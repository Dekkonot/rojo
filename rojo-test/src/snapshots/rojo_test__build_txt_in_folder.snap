---
created: "2019-08-08T00:40:50.026718300Z"
creator: insta@0.10.0
source: rojo-test/src/lib.rs
expression: contents
---
<roblox version="4">
  <Item class="Folder" referent="0">
    <Properties>
      <string name="Name">txt_in_folder</string>
    </Properties>
    <Item class="StringValue" referent="1">
      <Properties>
        <string name="Name">foo</string>
        <string name="Value"></string>
      </Properties>
    </Item>
  </Item>
</roblox>
